### ISSUE TITLE
SVG.js

### ISSUE THREAD

**Issue by Dominus77** [GHopperMSK/svgMap#14]
Привет!
Видел такую js библиотеку?
http://svgjs.com/
Почему бы не использовать её?)

**Comment by np25071984** [comment]
Да, видел. Я изначально 3D.js использовал - там простая работа с массивами данных и матрицами преобразования.
Вопрос в другом - зачем оно все нам? Вот что ты сейчас не можешь сделать на чистом js, чтобы приегнуть к чужим библиотекам?

**Comment by Dominus77** [comment]
Ну это просто к тому, что зачем изобретать велосипед?))
https://github.com/Dominus77/yii2-svg-widget

**Comment by np25071984** [comment]
Ну это смотря что называть велосипед. В проекте по факту нет ничего сложного. Вообще! Т.е. создали кондейнер из несколькоих дивов. Пропарсили json и запихнули в svg. Написали прару обробочиков - овер, клик и с замом поигрались. Все. Объем всего добра - пара сотенстрочек включая описание. О каком велосипеде речь?
Еще раз вопрос - а в чем проблема сделать все руками?

**Comment by Dominus77** [comment]
Так зачем делать то что уже сделано? Только ради самообразования)

**Comment by np25071984** [comment]
Вот это хороший вопрос. Что уже сделано?

**Comment by Dominus77** [comment]
В библиотеке заложен базовый функционал с дополнительными плюшками, бери и используй, адаптируй для своей задачи.

**Comment by np25071984** [comment]
Да, если она позволяет
1. создавать svg из json
2. кастомизировать внешний вид
3. кастомизировать обработчики событий
4. поддерживает кросброузерность

Может закрыть свой проект и юзать готовый.

**Comment by Dominus77** [comment]
> Да, если она позволяет
> 
>     создавать svg из json
>     кастомизировать внешний вид
>     кастомизировать обработчики событий
>     поддерживает кросброузерность
> 
> Может закрыть свой проект и юзать готовый.

В этом и заключается задача, научить библиотеку первому пункту. 
Всё остальное она умеет.
Ну и добавить свои плюшки)

**Comment by np25071984** [comment]
```
var draw = SVG('drawing');
draw.path('M0 0 H50 A20 20 0 1 0 100 50 v25 C50 125 0 85 0 85 z');
...

````

Вот тебе и будет тоже самое, что у нас сейчас.

**Comment by Dominus77** [comment]
Ну а я про что)

**Comment by Dominus77** [comment]
Организовать всё это дело в виджет и передавать параметры через php в том же массиве.

**Comment by np25071984** [comment]
В целом да, мы с тобой ни байна нового не привзнесли в этот мир.
У меня была цель посмотреть процесс изнутри. Сама идея, к сожалению, оказалась не новой и уже давно отработанной.
Что теперь делать - не понятно.В крупный ОпенСорс-проект падаться? ...

**Comment by np25071984** [comment]
> Организовать всё это дело в виджет и передавать параметры через php в том же массиве.

Мы же все понимаем, что здесь и виджет нафик никому не нужен. Это обычный враппер, который передает три твои параметра в js. Но это можно с тем-же успехом селать и напрямую.

**Comment by Dominus77** [comment]
Для yii2 вроде еще нет адаптации работы с данной библиотекой. Вот можно и адаптировать.

**Comment by np25071984** [comment]
у тебя телега есть?

**Comment by Dominus77** [comment]
Telegram?

**Comment by np25071984** [comment]
ну

**Comment by Dominus77** [comment]
есть

**Comment by np25071984** [comment]
) пуляй сюда ее. Будем в телеге писаться

**Comment by Dominus77** [comment]
для компа версия есть телеграма а то у меня на телефоне, а там не удобно как то))

**Comment by np25071984** [comment]
У меня все на компе

**Comment by np25071984** [comment]
В броузере прям

**Comment by Dominus77** [comment]
`@Dominus77`

**Comment by Dominus77** [comment]
Надо пошарить, тоже найти что нибудь для компа))

**Comment by Dominus77** [comment]
О, нашел для Firefox
